Lib/fontTools/subset/cff.py

```diff
@@ -104,37 +116,41 @@ def subset_glyphs(self, s):
 		font = cff[fontname]
 		cs = font.CharStrings
 
-		# Load all glyphs
-		for g in font.charset:
-			if g not in s.glyphs: continue
-			c, _ = cs.getItemAndSelector(g)
-
-		if cs.charStringsAreIndexed:
-			indices = [i for i,g in enumerate(font.charset) if g in s.glyphs]
-			csi = cs.charStringsIndex
-			csi.items = [csi.items[i] for i in indices]
-			del csi.file, csi.offsets
-			if hasattr(font, "FDSelect"):
-				sel = font.FDSelect
-				# XXX We want to set sel.format to None, such that the
-				# most compact format is selected. However, OTS was
-				# broken and couldn't parse a FDSelect format 0 that
-				# happened before CharStrings. As such, always force
-				# format 3 until we fix cffLib to always generate
-				# FDSelect after CharStrings.
-				# https://github.com/khaledhosny/ots/pull/31
-				#sel.format = None
-				sel.format = 3
-				sel.gidArray = [sel.gidArray[i] for i in indices]
-			cs.charStrings = {g:indices.index(v)
-					  for g,v in cs.charStrings.items()
-					  if g in s.glyphs}
+		if s.options.retain_gids:
+			for g in s.glyphs_emptied:
+				_empty_charstring(font, g, isCFF2=cff.major > 1, width=0)
```

Comment:
I think we want to pass private.defaultWidthX here.

Reply:
I don't understand.. i'm passing the cff font in as parameter, then for CFF 1.0 I'm getting it's private.defaultWidthX and compute/store the charstring width. I thought for CFF2 we never want to store the width.

Reply:
You're passing width=0, which requires storage in the charstring in CFF 1.0.  We don't care about width of the emptied glyphs, so we want to not store anything, which is equivalent to default width.  For CFF2 of course, there is no width whatsoever.